libs/synchronization/include/hpx/mutex.hpp

```diff
@@ -0,0 +1,21 @@
+//  Copyright (c) 2020 ETH Zurich
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once
+
+#include <hpx/synchronization.hpp>
+#include <hpx/thread_support.hpp>
+
+namespace hpx {
+    using hpx::lcos::local::call_once;
+    using hpx::lcos::local::mutex;
+    using hpx::lcos::local::no_mutex;
+    using hpx::lcos::local::once_flag;
+    using hpx::lcos::local::recursive_mutex;
+    using hpx::lcos::local::shared_mutex;
+    using hpx::lcos::local::spinlock;
+    using hpx::util::unlock_guard;
+}    // namespace hpx
```

Comment:
Another note to self: we need to implement `timed_mutex` and `recursive_timed_mutex`.

Reply:
And a note to myself before merging this: we already have `timed_mutex` so add it (we don't have `recursive_timed_mutex`).